--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-rk.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-rk.c
@@ -1704,8 +1705,11 @@ static int rk_gmac_clk_init(struct plat_
 		bsp_priv->clks[i].id = rk_clocks[i];
 
 	if (phy_iface == PHY_INTERFACE_MODE_RMII) {
-		for (j = 0; j < ARRAY_SIZE(rk_rmii_clocks); j++)
+		for (j = 0; j < ARRAY_SIZE(rk_rmii_clocks); j++) {
+			if(bsp_priv->clock_input && !strcmp(rk_rmii_clocks[j], "clk_mac_refout"))
+				continue;
 			bsp_priv->clks[i++].id = rk_rmii_clocks[j];
+		}
 	}
 
 	ret = devm_clk_bulk_get_optional(dev, bsp_priv->num_clks,
@@ -2028,6 +2067,7 @@ static int rk_gmac_probe(struct platform
 	 */
 	if (!plat_dat->has_gmac4)
 		plat_dat->has_gmac = true;
+	plat_dat->flags |= STMMAC_FLAG_SPH_DISABLE;
 	plat_dat->fix_mac_speed = rk_fix_speed;
 
 	plat_dat->bsp_priv = rk_gmac_setup(pdev, plat_dat, data);
